esmvalcore/_recipe.py

```diff
@@ -401,6 +402,11 @@ def _get_correct_fx_file(variable, fx_varname, config_user):
         elif fx_varname == 'sftgif':
             fx_var = _add_fxvar_keys({'short_name': fx_varname, 'mip': 'fx'},
                                      var)
+    else:
+        raise NotImplementedError("Project %s not \
```

Comment:
Change to RecipeError and use an f-string to format the error message instead of % formatting